modules/apps/object/object-test/src/testFunctional/tests/ObjectSalesforce.testcase

```diff
@@ -21,4 +21,84 @@ definition {
 		}
 	}
 
+	@description = "LPS-185427 - Verify CRUD functionality with an imported custom object using Salesforce as the storage type."
+	@priority = 5
+	test AssertCRUDWithImportedCustomObject {
+		property portal.acceptance = "true";
+
+		task ("Given portal configured with Salesforce") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Configuration",
+				panel = "Control Panel",
+				portlet = "Instance Settings");
+
+			SystemSettings.gotoConfiguration(
+				configurationCategory = "Third Party",
+				configurationName = "Salesforce Integration",
+				configurationScope = "Site Scope");
+
+			ObjectSalesforce.configureSalesforce();
+		}
+
+		task ("And given an imported custom object whose external reference code matches a Salesforce custom object") {
+			ObjectExportImport.importObject(
+				objectFileName = "ImportedObjectWithSalesforceStorageType",
+				objectName = "PoshiTest");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "PoshiTest");
+		}
+
+		task ("When the user navigates to the object") {
+			ObjectAdmin.goToCustomObject(objectName = "PoshiTest");
+		}
+
+		task ("Then the user is able to create") {
+			ObjectAdmin.addObjectStringFieldEntryViaUI(entryValue = "New Entry");
+
+			ObjectAdmin.goToCustomObject(objectName = "PoshiTest");
+
+			AssertElementPresent(
+				key_entry = "New Entry",
+				locator1 = "ObjectPortlet#ENTRY_VALUE");
+		}
+
+		task ("And then the user is able to read") {
```

Comment:
Let's assert this step first so that the rest of the steps interact with the same object entry